examples/with-cookie-auth/lib/auth.js

```diff
@@ -0,0 +1,97 @@
+import Router from 'next/router'
+import axios from 'axios'
+
+axios.defaults.withCredentials = true
+
+const WINDOW_USER_SCRIPT_VARIABLE = `__USER__`
+
+const decode = ({token}) => {
+  if (!token) {
+    return {}
+  }
+  const {email, type} = token || {}
+  return {user: {email, type}}
+}
+
+export const getUserScript = (user) => {
+  const json = JSON.stringify(user)
+  return `${WINDOW_USER_SCRIPT_VARIABLE} = ${json};`
+}
+
+export const getServerSideToken = (req) => {
+  const {signedCookies} = req
+
+  if (!signedCookies) {
+    return {}
+  }
+  try {
+    return decode(signedCookies)
+  } catch (parseError) {
+    return {}
+  }
+}
+
+export const getClientSideToken = () => {
+  if (typeof window !== 'undefined') {
+    const user = window[WINDOW_USER_SCRIPT_VARIABLE] || {}
+    return {user}
+  }
+  return {user: {}}
+}
+
+const getRedirectPath = (userType) => {
+  switch (userType) {
+    case 'authenticated':
+      return '/profile'
+    default:
+      return '/login'
+  }
+}
+
+const redirect = (res, path) => {
+  if (res) {
+    res.redirect(302, path)
```

Comment:
`res.redirect` is not something officially supported, it comes from Express, this should use the location header.